source/common/grpc/codec.h

```diff
@@ -56,40 +88,28 @@ class Decoder {
   bool hasBufferedData() const { return state_ != State::FH_FLAG; }
 
 private:
-  // Wire format (http://www.grpc.io/docs/guides/wire.html) of GRPC data frame
-  // header:
-  //
-  // -----------------------------------------------------------------------
-  // |R|R|R|R|R|R|R|R|C|      L     |      L     |      L     |      L     |
-  // -----------------------------------------------------------------------
-  //    Flag (1 byte)                Message Length (4 bytes)
-  //
-  // A fixed header consists of five bytes.
-  // The first byte is the Flag. The last one "C" bit indicates if the message
-  // is compressed or not (0 is uncompressed, 1 is compressed). The other seven
-  // "R" bits are reserved for future use.
-  // The next four "L" bytes represent the message length in BigEndian format.
-  enum class State {
-    // Waiting for decoding the flags (1 byte) of the GRPC data frame.
-    FH_FLAG,
-    // Waiting for decoding the 1st byte of the length (4 bytes in total) of the
-    // GRPC data frame.
-    FH_LEN_0,
-    // Waiting for decoding the 2nd byte of the length (4 bytes in total) of the
-    // GRPC data frame.
-    FH_LEN_1,
-    // Waiting for decoding the 3rd byte of the length (4 bytes in total) of the
-    // GRPC data frame.
-    FH_LEN_2,
-    // Waiting for decoding the 4th byte of the length (4 bytes in total) of the
-    // GRPC data frame.
-    FH_LEN_3,
-    // Waiting for decoding the data.
-    DATA,
-  };
-
   State state_;
   Frame frame_;
 };
+
+class FrameInspector {
```

Comment:
Optimally we would have fuzzing for all of the gRPC frame code that we have within Envoy. Can we potentially look at this as a follow up? cc @asraa @htuch

Reply:
Sounds good.